### PR TITLE
[CRD][2/n] Update from CRD v1alpha1 to v1

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

This PR is the step 2 of https://github.com/ray-project/kuberay/issues/1209#issuecomment-1758211016.

* https://github.com/ray-project/kuberay/pull/1482/commits/76fe63fe663ae31c19c215370e2baaa805d2f26f
  * Set v1 as the storage version 
* https://github.com/ray-project/kuberay/pull/1482/commits/68177930234a28f379d040ecae9d86f16c4fe5eb
  * Update `update-codegen.sh` from v1alpha1 to v1.
  * Run `./hack/update-codegen.sh` to create lister / informer / client for v1.
  * Delete v1alpha1 related directories in `pkg/client`.
  * Create `apis/ray/v1/register.go`.
* https://github.com/ray-project/kuberay/pull/1482/commits/c48e91ea7d2f6b7e67adf69183ffee25de6defce
  * Replace all occurrences of `github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1` in `ray-operator/controllers/ray` with `github.com/ray-project/kuberay/ray-operator/apis/ray/v1`. 
* https://github.com/ray-project/kuberay/pull/1482/commits/fd6dd5890f28db2c56f6aae78e90398b2db8530d
  * Update KubeRay API server.
* https://github.com/ray-project/kuberay/pull/1482/commits/868f328a0d63bf76275e5c10d79512db461dd4ac
  * Replace all occurrences of `rayv1alpha1` in `ray-operator/controllers/ray` with `rayv1`.
* https://github.com/ray-project/kuberay/pull/1482/commits/07b659d71f566288dbfaf7523e42276c63dd4eb6
  * Revert a change based on https://github.com/ray-project/kuberay/pull/1482#discussion_r1356534773. This PR will not upgrade KubeRay API server's version. cc @blublinsky @z103cb 
* https://github.com/ray-project/kuberay/pull/1482/commits/7657d0e98eb6cb81b885b09c93895e428246ca95
  * The latest release v1.0.0-rc.0 does not support CRD v1, so the sample YAML tests fail consistently. This commit comments the tests out, and will uncomment them after v1.0.0-rc.1 is out.

## Related issue number
#1209
#1481 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: apiserver/cmd/main.go
Comment (z103cb) [inline apiserver/cmd/main.go:131]: This change is meaningless without changes in the `.proto` files. I think that apiserver API might not be 'v1' ready. Do you want to make the changes to the API server rest path now ? I think we can move the Kuberay API server paths independent of the underlying kuberay operator clients.

cc: blublinsky

**Comment by z103cb** [comment]
@kevin85421,
I think you might want to rebase main unto this PR. I have tried running `make test` in the apiserver directory and got this error

```text
--- FAIL: TestBuildVolumes (0.00s)
    --- FAIL: TestBuildVolumes/configmap_test (0.00s)
        cluster_test.go:377: failed for configmap test ..., got [{configMap {nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil &ConfigMapVolumeSource{LocalObjectReference:LocalObjectReference{Name:my-config-map,},Items:[]KeyToPath{KeyToPath{Key:key2,Path:path2,Mode:nil,},KeyToPath{Key:key1,Path:path1,Mode:nil,},},DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil nil nil}}], expected [{configMap {nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil nil &ConfigMapVolumeSource{LocalObjectReference:LocalObjectReference{Name:my-config-map,},Items:[]KeyToPath{KeyToPath{Key:key1,Path:path1,Mode:nil,},KeyToPath{Key:key2,Path:path2,Mode:nil,},},DefaultMode:nil,Optional:nil,} nil nil nil nil nil nil nil nil nil nil}}]
FAIL
```

There is was some test flakiness that @Blublinsky has fixed in his latest merged PR.

File: apiserver/cmd/main.go
Comment (blublinsky) [reply apiserver/cmd/main.go:131]: To move to v1 we need to modify .proto files, for example:
````
service ClusterService {
  // Creates a new Cluster.
  rpc CreateCluster(CreateClusterRequest) returns (Cluster) {
    option (google.api.http) = {
      post: "/apis/v1alpha2/namespaces/{namespace}/clusters"
      body: "cluster"
    };
  }

  // Finds a specific Cluster by ID.
  rpc GetCluster(GetClusterRequest) returns (Cluster) {
    option (google.api.http) = {
      get: "/apis/v1alpha2/namespaces/{namespace}/clusters/{name}"
    };
  }

  // Finds all Clusters in a given namespace. Supports pagination, and sorting on certain fields.
  rpc ListCluster(ListClustersRequest) returns (ListClustersResponse) {
    option (google.api.http) = {
      get: "/apis/v1alpha2/namespaces/{namespace}/clusters"
    };
  }

  // Finds all Clusters in all namespaces. Supports pagination, and sorting on certain fields.
  rpc ListAllClusters(ListAllClustersRequest) returns (ListAllClustersResponse) {
    option (google.api.http) = {
      get: "/apis/v1alpha2/clusters"
    };
  }

  // Deletes an cluster without deleting the cluster's runs and jobs. To
  // avoid unexpected behaviors, delete an cluster's runs and jobs before
  // deleting the cluster.
  rpc DeleteCluster(DeleteClusterRequest) returns (google.protobuf.Empty) {
    option (google.api.http) = {
      delete: "/apis/v1alpha2/namespaces/{namespace}/clusters/{name}"
    };
  }
}
````
and replace `v1alpha2` with `v1`

We can do it now, but its a breaking change

**Comment by z103cb** [comment]
I have ran a few of the examples from the README.md, so far this PR is LGTM. I will approve it once the review comments are disposed of and the PR is no long in WIP state.

**Comment by kevin85421** [comment]
> @z103cb There is was some test flakiness that @blublinsky has fixed in his latest merged PR.

My branch has the commit #1461. 

Screen Shot 2023-10-12 at 12 52 02 PM

File: apiserver/cmd/main.go
Comment (kevin85421) [reply apiserver/cmd/main.go]: I reverted the change https://github.com/ray-project/kuberay/pull/1482/commits/07b659d71f566288dbfaf7523e42276c63dd4eb6.

**Review by hongchaodeng — APPROVED** [review]
LGTM
Let's merge this after @blublinsky's approval.

**Review by z103cb — APPROVED** [review]
LGTM

**Comment by z103cb** [comment]
> > @z103cb There is was some test flakiness that @blublinsky has fixed in his latest merged PR.
> 
> My branch has the commit #1461.
> 
> Screen Shot 2023-10-12 at 12 52 02 PM


Ok. Will need to fix the flaky test then, in a separate PR (this should be merged).

**Comment by kevin85421** [comment]
Thanks, @hongchaodeng @z103cb @blublinsky!

**Pull request merged**
